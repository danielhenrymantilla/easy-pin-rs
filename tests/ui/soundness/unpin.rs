#![allow(unused)]

use ::easy_pin::{
    easy_pin,
    PinSensitive,
};
use ::core::{
    ptr::NonNull,
    marker::Unpin,
};

#[easy_pin]
pub
struct SelfReferential {
    #[transitively_pinned]
    string: PinSensitive<String>,

    #[unpinned]
    at_string: NonNull<String>,
}

impl Unpin for SelfReferential {}

fn main ()
{}
