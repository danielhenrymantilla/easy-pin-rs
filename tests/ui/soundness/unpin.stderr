error[E0119]: conflicting implementations of trait `std::marker::Unpin` for type `SelfReferential`:
  --> $DIR/unpin.rs:12:1
   |
12 | #[easy_pin]
   | ^^^^^^^^^^^ conflicting implementation for `SelfReferential`
...
22 | impl Unpin for SelfReferential {}
   | ------------------------------ first implementation here

For more information about this error, try `rustc --explain E0119`.
